### ISSUE TITLE
Incorrect rendering of reStructuredText hyperlinks in `TypeIs` section of Type Narrowing documentation page

### ISSUE THREAD

**Issue by neuroevolutus** [python/mypy#18056]
**Documentation**

As stated in the title, a couple of the reStructuredText hyperlinks on the [Type Narrowing](https://mypy.readthedocs.io/en/stable/type_narrowing.html#typeis) page are not being correctly converted into HTML. The latter two hyperlinks of the following reStructuredText source are at fault:

```reStructuredText
A `TypeIs narrowing function <https://typing.readthedocs.io/en/latest/spec/narrowing.html#typeis>`_
allows you to define custom type checks that can narrow the type of a variable
in `both the if and else <https://docs.python.org/3.13/library/typing.html#typing.TypeIs>`_
branches of a conditional, similar to how the built-in isinstance() function works.

TypeIs is new in Python 3.13 — for use in older Python versions, use the backport
from `typing_extensions <https://typing-extensions.readthedocs.io/en/latest/>`_
```

**Comment by brianschubert** [comment]
Thanks! This was fixed in #18028. You should be able to see the corrected markup if you switch the docs version from “stable” to “latest”

**Comment by neuroevolutus** [comment]
That makes sense now, 😅. Thanks for letting me now!
